lib/phoenix/sync/write.ex

```diff
@@ -0,0 +1,970 @@
+defmodule Phoenix.Sync.Write do
+  @moduledoc """
+  Provides [optimistic write](https://electric-sql.com/docs/guides/writes)
+  support for Phoenix- or Plug-based apps.
+
+  The client uses some local database and syncs changes into this database from
+  the server using [Electric](https://electric-sql.com/).
+
+  Local writes are performed on the local database using
+  [TanStack/optimistic](https://github.com/TanStack/optimistic) which POSTs
+  transactions to the Elixir/Phoenix application.
+
+  A transaction is a list of "mutations", a series of `INSERT`, `UPDATE` and
+  `DELETE` operations. These operations are then written to the server's
+  database via the functions in this module and the transaction id of this
+  write is returned to the client.
+
+  When the client receives this transaction id back through it's Electric sync
+  stream then the client knows that it's up-to-date with the server.
+
+  ## Client Code
+
+  See [TanStack/optimistic](https://github.com/TanStack/optimistic) for more information.
```

Comment:
This needs to be presented as one example approach. I'm also not sure it belongs here in the moduledoc?

Reply:
Maybe not the full integration code -- likely to just fall out of date anyway. Do think it's worth pointing to front-end libs here though. Maybe just a list